WordPress/Classes/ViewRelated/Reader/ReaderSubscriptionsView.swift

```diff
@@ -0,0 +1,191 @@
+import SwiftUI
+import WordPressUI
+
+struct ReaderSubscriptionsView: View {
+    @FetchRequest(
+        sortDescriptors: [SortDescriptor(\.title, order: .forward)],
+        predicate: NSPredicate(format: "following = YES")
+    )
+    private var subscriptions: FetchedResults<ReaderSiteTopic>
+
+    @State private var searchText = ""
+    @State private var isAddingSubscription = false
+
+    @State private var searchResults: [ReaderSiteTopic] = []
+
+    var isShowingSearchResuts: Bool { !searchText.isEmpty }
+
+    var onSelection: (_ subscription: ReaderSiteTopic) -> Void = { _ in }
+
+    private var visibleSubscriptions: any RandomAccessCollection<ReaderSiteTopic> {
+        if searchText.isEmpty {
+            return subscriptions
+        } else {
+            return searchResults
+        }
+    }
+
+    var body: some View {
+        Group {
+            if subscriptions.isEmpty {
+                emptyStateView
+            } else {
+                main
+            }
+        }
+        .navigationTitle(Strings.title)
+    }
+
+    private var emptyStateView: some View {
+        EmptyStateView {
+            Label(Strings.title, systemImage: "doc.text.magnifyingglass")
+        } description: {
+            Text(Strings.emptyStateDetails)
+        } actions: {
+            buttonAddSubscription
+                .fixedSize(horizontal: true, vertical: /*@START_MENU_TOKEN@*/true/*@END_MENU_TOKEN@*/)
+                .padding(EdgeInsets(top: 12, leading: 8, bottom: 8, trailing: 10))
+                .background(Color(.secondarySystemBackground).opacity(0.8))
+                .clipShape(RoundedRectangle(cornerRadius: 12))
+        }
+    }
+
+    private var main: some View {
+        List {
+            if !isShowingSearchResuts {
+                buttonAddSubscription
+                    .listRowSeparator(.hidden)
+                ForEach(subscriptions, id: \.objectID, content: makeSubscriptionCell)
+                    .onDelete(perform: delete)
+            } else {
+                ForEach(searchResults, id: \.objectID, content: makeSubscriptionCell)
+                    .onDelete(perform: delete)
+            }
+        }
+        .listStyle(.plain)
+        .searchable(text: $searchText, placement: .navigationBarDrawer)
+        .onReceive(subscriptions.publisher) { _ in
+            if !searchText.isEmpty {
+                reloadSearchResults(searchText: searchText)
+            }
+        }
+        .onChange(of: searchText) {
+            reloadSearchResults(searchText: $0)
+        }
+        .toolbar {
+            EditButton()
+        }
+    }
+
+    private func makeSubscriptionCell(for site: ReaderSiteTopic) -> some View {
+        Button {
+            onSelection(site)
+        } label: {
+            ReaderSubscriptionCell(site: site, onDelete: delete)
+        }
+    }
+
+    private var buttonAddSubscription: some View {
```

Comment:
What do you think putting this button at the navigation bar, which I think is a pretty common pattern?

Reply:
I added it to the list with the intention to make it more discoverable that you can add any types of sites using RSS. I could also add it to the navigation bar, but it just require a bit more effort to re-wire the popover 😬

Reply:
I refactored it to a separate `ReaderAddSubscriptionButton` and added it to the new empty state view and navigation bar.